src/coffea/nanoevents/methods/vector.py

```diff
@@ -795,6 +655,12 @@ def negative(self):
             behavior=self.behavior,
         )
 
+    @awkward.mixin_class_method(numpy.divide, {numbers.Number})
+    def divide(self, other):
+        """Divide this vector by a scalar elementwise using its cartesian components
+        This is realized by using the multiplication functionality"""
+        return self.multiply(1 / other)
+
```

Comment:
I've added the divide methods in `PtEtaPhiMLorentzVector` and `PtEtaPhiELorentzVector` classes to preserve the coordinate names. I have not trimmed down the `multiply` and `negative` methods of these classes for the exact same reason.

Vector automatically canonicalizes coordinate names from momentum synonyms to generic ones, and I think we would not want that here.

Reply:
By canonicalize do you mean that it rewrites the underlying awkward RecordArray field names?